### DISCUSSION TITLE
v0.9.1 - InstaFlow TextToImage

### DISCUSSION THREAD

**Discussion by saddam213** [saddam213/OnnxStack#49] [General]
Discussion for feedback while we implement InstaFlow

Initial support for InstaFlow 1 Step inference [InstaFlow Github](https://github.com/gnobitab/InstaFlow)

ONNX converted model here: 
[InstaFlow-0.9B-ONNX](https://huggingface.co/TheyCallMeHex/InstaFlow-0.9B-ONNX)
[2-Rectified-Flow-ONNX](https://huggingface.co/TheyCallMeHex/2-Rectified-Flow-ONNX)

Preview Build 
https://github.com/saddam213/OnnxStack/releases/tag/v0.9.1-pre

**Comment by AshD** [comment]
Cool. This looks amazing!

Is the license for this only for Non Commercial use?
https://github.com/gnobitab/InstaFlow/tree/main/code

Thanks,
Ash

**Comment by Amin456789** [comment]
nice!
@AshD it will be great if u convert instaflow upcoming models to fp16 too!

**Comment by AshD** [comment]
I can try that once they are released. Ping me with the Model url. 

Regards,
Ash

**Comment by saddam213** [comment]
https://huggingface.co/XCLiu/instaflow_0_9B_from_sd_1_5
https://huggingface.co/XCLiu/2_rectified_flow_from_sd_1_5

First is the 1step, Second is their low step, LCM type model

**Comment by AshD** [comment]
Got an error. **AttributeError: module diffusers has no attribute RectifiedFlowPipeline**

Need to do some more research here to modify olive to support this model. Waiting on the Olive team to fix couple of my open issues.

Regards,
Ash

```
python stable_diffusion.py --model_id="XCLiu/instaflow_0_9B_from_sd_1_5" --optimize
WARNING: XCLiu/instaflow_0_9B_from_sd_1_5 is not an officially supported model for this example and may not work as expected.
Download stable diffusion PyTorch pipeline...
model_index.json: 100%|███████████████████████████████████████████████████████████████████████| 575/575 [00:00<?, ?B/s]
Traceback (most recent call last):
  File "C:\AI\research\olive\olive\examples\directml\stable_diffusion\stable_diffusion.py", line 355, in <module>
    optimize(args.model_id, unoptimized_model_dir, optimized_model_dir)
  File "C:\AI\research\olive\olive\examples\directml\stable_diffusion\stable_diffusion.py", line 195, in optimize
    pipeline = DiffusionPipeline.from_pretrained(base_model_id, torch_dtype=torch.float32)
  File "C:\AI\research\olive\.venv\lib\site-packages\diffusers\pipelines\pipeline_utils.py", line 1078, in from_pretrained
    cached_folder = cls.download(
  File "C:\AI\research\olive\.venv\lib\site-packages\diffusers\pipelines\pipeline_utils.py", line 1754, in download
    pipeline_class = _get_pipeline_class(
  File "C:\AI\research\olive\.venv\lib\site-packages\diffusers\pipelines\pipeline_utils.py", line 376, in _get_pipeline_class
    pipeline_cls = getattr(diffusers_module, class_name)
  File "C:\AI\research\olive\.venv\lib\site-packages\diffusers\utils\import_utils.py", line 703, in __getattr__
    raise AttributeError(f"module {self.__name__} has no attribute {name}")
AttributeError: module diffusers has no attribute RectifiedFlowPipeline
```

**Reply by Amin456789** [reply]
thanks for ur hard work

**Comment by AshD** [comment]
Quick update. I made progress but I am blocked on this issue
https://github.com/microsoft/Olive/issues/757

**Comment by AshD** [comment]
Good news. I rebuilt my python env and reinstalled everything and it is working.
https://huggingface.co/softwareweaver/Rectified-Flow-Olive-Onnx
https://huggingface.co/softwareweaver/InstaFlow-0.9B-Olive-Onnx

@saddam213 Let me know if you have any issues.

**Reply by Amin456789** [reply]
goodie, thank u!
